### PR TITLE
Add account registration and activation

### PR DESCRIPTION
**Pull request by Sheikah45**

### CODE REVIEW

File: src/main/kotlin/com/faforever/userservice/ui/view/registration/ActivateView.kt [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+@Route("/register/activate", layout = CardLayout::class)
+class ActivateView(private val registrationService: RegistrationService) :
+    CompactVerticalLayout(), BeforeEnterObserver {
+
+    companion object {
+        class PasswordConfirmation {
+            var password: String = ""
+            var confirmedPassword: String = ""
+        }
```
Comment (Brutus5000) [inline src/main/kotlin/com/faforever/userservice/ui/view/registration/ActivateView.kt:36]: The wrapping companion object should be unnecessary

File: src/main/kotlin/com/faforever/userservice/ui/view/registration/ActivateView.kt [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        binder.addStatusChangeListener { submit.isEnabled = it.binder.isValid }
+    }
+
+    private fun activate() {
+        val validationStatus = binder.validate()
+        if (validationStatus.hasErrors()) {
+            return
+        }
+
+        val ipAddress = IpAddress(VaadinSession.getCurrent().browser.address)
```
Comment (Brutus5000) [inline src/main/kotlin/com/faforever/userservice/ui/view/registration/ActivateView.kt:104]: Does this take proxy/cloudflare headers into account?

Comment (Sheikah45) [reply src/main/kotlin/com/faforever/userservice/ui/view/registration/ActivateView.kt:104]: No it doesn't so need to update that

**Comment by Brutus5000** [comment]
Ah jesus spotless broke on the last mile

**Comment by Sheikah45** [comment]
I really need to align my intellij with spotless for this project

**Pull request merged**